llvm/tools/llvm-exegesis/lib/Assembler.cpp

```diff
@@ -323,10 +324,8 @@ Error assembleToStream(const ExegesisTarget &ET,
   TPC->printAndVerify("After ExegesisTarget::addTargetSpecificPasses");
```

Comment:
But the verifier added here and line 321 would still abort upon verification failure. Should we also return Error for these two places?

Reply:
Those only run when passing `-verify-machineinstrs` on the command line. It's a lot harder to return an error there rather than just abort given the calls are to the MachineVerifier are nested within the pass pipeline, and the checks need to be explicitly enabled by the user on the command line where I think aborting is probably the most reasonable course of action.

Reply:
Right, I was thinking about replacing `TPC->printAndVerify(...)` with manual MachineFunction verification like you did in line 340. But I don't have a strong opinion on this.

> and the checks need to be explicitly enabled by the user on the command line

Just FYI I think `LLVM_ENABLE_EXPENSIVE_CHECKS` will also enable those by default.

Reply:
Yeah. I think for now this is fine. It accomplishes what we need downstream and isn't too invasive. I'll revisit down the line if we need something else/we need to clean up upstream.